File: Design-Patterns-Using-Swift/SOLID/LiskovSubstitutionPrinciple.swift

// Animal class
class Animal {
    var name: String
    
    init(name: String) {
        self.name = name
    }
    
    func makeSound() {
        print("Animal sound")
    }
}

// Dog class
class Dog: Animal {
    override func makeSound() {
        print("woof woof!")
    }
}

// Cat class
class Cat: Animal {
    override func makeSound() {
        print("meow!")
    }
}

class Rectangle {
    private var height: Int
    private var width: Int

    init(height: Int, width: Int) {
        self.height = height
        self.width = width
    }

    func setWidth(_ width: Int) {
        self.width = width
    }

    func setHeight(_ height: Int) {
        self.height = height
    }

    func getWidth() -> Int {
        return width
    }

    func getHeight() -> Int {
        return height
    }
}

class Square: Rectangle {
    init(sideLength: Int) {
        super.init(height: sideLength, width: sideLength)
    }

    override func setWidth(_ width: Int) {
        super.setWidth(width)
        super.setHeight(width)
    }

    override func setHeight(_ height: Int) {
        super.setWidth(height)
        super.setHeight(height)
    }
}

func testRect(rect: Rectangle) {
    rect.setHeight(10)
    rect.setWidth(20)
    if 200 == rect.getHeight() * rect.getWidth() {
        print("success")
    } else {
        print("failure")
    }
}

// Client Code
let r = Rectangle(height: 10, width: 20)
testRect(rect: r)
let s = Square(sideLength: 10)
testRect(rect: s)

/*
success
failure
*/

File: Design-Patterns-Using-JavaScript/SOLID/LiskovSubstitutionPrinciple.js
class Animal {
    constructor(name) {
        this.name = name;
    }

    makeSound() {
        console.log("Animal sound");
    }
}

class Dog extends Animal {
    constructor(name) {
        super(name);
    }

    makeSound() {
        console.log("woof woof!");
    }
}

class Cat extends Animal {
    constructor(name) {
        super(name);
    }

    makeSound() {
        console.log("meow!");
    }
}

class Rectangle {
    constructor(l, w) {
        this.height = l;
        this.width = w;
    }

    setWidth(w) {
        this.width = w;
    }

    setHeight(h) {
        this.height = h;
    }

    getWidth() {
        return this.width;
    }

    getHeight() {
        return this.height;
    }
}

class Square extends Rectangle {
    constructor(l) {
        super(l, l);
    }

    setWidth(w) {
        super.setWidth(w);
        super.setHeight(w);
    }

    setHeight(h) {
        super.setWidth(h);
        super.setHeight(h);
    }
}

function testRect(rect) {
    rect.setHeight(10);
    rect.setWidth(20);
    if (200 === rect.getHeight() * rect.getWidth()) {
        console.log("success");
    } else {
        console.log("failure");
    }
}

// Client code
const r = new Rectangle(10, 20);
testRect(r);
const s = new Square(10);
testRect(s);

/*
success
failure
*/